runelite-client/src/main/java/net/runelite/client/plugins/poison/PoisonPlugin.java

```diff
@@ -0,0 +1,219 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.poison;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.text.MessageFormat;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import net.runelite.client.eventbus.Subscribe;
+import com.google.inject.Provides;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import javax.inject.Inject;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.SpriteID;
+import net.runelite.api.VarPlayer;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.ColorUtil;
+
+@PluginDescriptor(
+	name = "Poison",
+	description = "Tracks current damage values for Poison and Venom",
+	tags = {"combat", "poison", "venom"}
+)
+@Slf4j
+public class PoisonPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private PoisonOverlay poisonOverlay;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private SpriteManager spriteManager;
+
+	@Inject
+	private PoisonConfig config;
+
+	@Provides
+	PoisonConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(PoisonConfig.class);
+	}
+
+	private boolean envenomed = false;
+
+	@Getter
+	private int lastDamage = 0;
+
+	private PoisonInfobox infobox;
+
+	private Instant poisonNaturalCure = Instant.now();
+	private Instant nextPoisonTick = Instant.now();
+
+	static int POISON_TICK_MILLIS = 18200;
+	private static int VENOM_THRESHOLD = 1000000;
+	private static int VENOM_MAXIUMUM_DAMAGE = 20;
+
+	private int lastValue = -1;
+
+	@Override
+	public void startUp() throws Exception
+	{
+		overlayManager.add(poisonOverlay);
+	}
+
+	@Override
+	public void shutDown() throws Exception
+	{
+		overlayManager.remove(poisonOverlay);
```

Comment:
You should be cleaning up ALL data in your plugin here + removing the infoboxes

Reply:
I've found 1 other piece of data that needs to be cleared up, otherwise everything else is evaluated before being used.